src/channel/BUILD.gn

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/chip.gni")
+
+static_library("channel") {
+  output_name = "libChannelAPI"
+
+  sources = [
+    "Channel.cpp",
+    "Channel.h",
+    "ChannelContext.cpp",
+    "ChannelContext.h",
+    "ChannelMgr.cpp",
+    "ChannelMgr.h",
```

Comment:
We are typically adding folder to the include file names. So `channel/Manager.h` should be fine. Any thoughts on class names? Correct namespacing will help, but it can still collide if the users (e.g. other CPP files) have `using <namespace>` directives in the code.

Reply:
Updated the name to `Manager.cpp/.h'